### ISSUE TITLE
Proposal: Add marketplace link to auto-generated header

### ISSUE THREAD

**Issue by mrlacey** [madskristensen/VsixSynchronizer#14]
I think it would be helpful to include the marketplace link in the header of the generated files.

I'm using this in a project where we want to help encourage community contributions by making it as easy as possible for people to contribute.
Having an extension installed as a requirement is a concern, but adding something to the code to clarify what's needed and where to get it from helps lessen this concern.

So, instead of having a header like this (the current behavior):
```
// ------------------------------------------------------------------------------
// <auto-generated>
//     This file was generated by VSIX Synchronizer
// </auto-generated>
// ------------------------------------------------------------------------------
```

It would look like this:

```
// ------------------------------------------------------------------------------
// <auto-generated>
//     This file was generated by VSIX Synchronizer
//     Available from https://marketplace.visualstudio.com/items?itemName=MadsKristensen.VsixSynchronizer64
// </auto-generated>
// ------------------------------------------------------------------------------
```

If linking to one of the 64 **OR** 32-bit versions is a concern, it could include a link to a marketplace search that returns both:
https://marketplace.visualstudio.com/search?term=%22vsix%20synchronizer%22&target=VS

I'll happily submit a PR for this change, but I recognize the benefit of discussing this idea first.

**Comment by madskristensen** [comment]
That's a great idea. A link to the search page might be better

**Comment by astrohart** [comment]
@madskristensen I agree, I am all for this.  The search page is fine.  Links -- whether to the search page or the More Info page (the first URL referenced above) are always problematical because you can't guarantee that either link won't eventually break due to, say, a query string syntax change imposed by the authors of the website.

**Comment by mrlacey** [comment]
Or how about a link to this repo and the addition of marketplace links in the readme?

**Comment by astrohart** [comment]
@mrlacey I think we are overthinking things just a tad.  One thing about the "More Info URL," as it is called, meaning specifically, the URL `https://marketplace.visualstudio.com/items?itemName=MadsKristensen.VsixSynchronizer64`, is there is a so-called service that is exposed by Visual Studio to this extension that provides the capability of looking up information about VS Marketplace extensions, including this URL(s).  It's included in the metadata about extensions that VS itself provides from the Marketplace from an internal service inside `devenv.exe` and/or its DLLs that it hooks into.

The file you are talking about, if my understanding is correct, is auto-generated on the fly every time there is a build (I think, I am not 100% sure), so if the link breaks, you just right click on the `.vsixmanifest` file, and say **Run Custom Tool**, and this extension should (again, I am not 100% sure because I have not actually looked at a very large subset of this code yet) re-generate the file with the most up-to-date link, assuming that it's written to perform the lookup as described above.

Also, I think, and again, I am not 100% sure, that 64-bit or 32-bit URLs can be requested from the internal Visual Studio service via input parameters to the method call.  That is my speculation, without having looked at this extension's code in-depth.  If that is the case, there is a simple algorithm to follow (pseudocode):

```
if VS Version >= 2022 then
    request 64-bit marketplace page URL from VS service
    write to the auto-generated file
else
    request 32-bit marketplace page URL from VS service
    write to the auto-generated file
 ```

This is an easy choice because all versions of `devenv.exe` 2019 and lower are 32-bit processes, whereas it's a 64-bit process in 2022 and above.

If the bitness of Windows is a concern, there is a way to, programmatically, determine that bitness from the system Registry.  However, since this is a Visual Studio _extension_, the bitness of `devenv.exe` should be the determining factor.

@madskristensen Can you correct me if I am obscenely wrong on this?  I am just being off the cuff here, but you're the one who has worked with this particular extension's code base the most, so your input is valued.  I don't want to say things that are way off base.

**Comment by mrlacey** [comment]
So, after hours of trying to test this and failing to get it to debug (including rewatching https://www.youtube.com/watch?v=w_ntVZb_26M) I worked out why I couldn't test it in VS2019.

The manifest in the project is configured to only support VS2022!!!

@madskristensen Is the intention that this extension only now works with VS2022 as that's what the code is configured to do?
If so, there's no need to have version detection and the inclusion of the appropriate link.

If the plan is to just support VS2022 and above going forward, I can remove my version selection code and raise the PR.
If you do want to support (& update) the VS2019 & VS2022 versions, how do you plan to manage this? (Given the VS2019 references have been removed from the codebase and there's no separate branch.)

**Comment by madskristensen** [comment]
Yes, only VS 2022 and it's a separate marketplace entry
